### PR TITLE
fix: endpoint fix for tune destination

### PR DESCRIPTION
**Pull request by aanshi07**
## What are the changes introduced in this PR?

Taking subdomain as an input and other query parameters and updating the endpoint accordingly.

## What is the related Linear task?

Resolves INT-282

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added in **new readability format**?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

File: src/v0/destinations/tune/transform.js [hunk 1]
```diff
@@ -43,10 +50,13 @@ const responseBuilder = (message, { Config }) => {
       ...mapPropertiesWithNestedSupport(message, properties, advUniqueIdHashMap),
     };
 
+    const endpoint = getTuneEndpoint(subdomain);
+    const queryString = toQueryString(params);
+    const finalEndpoint = queryString ? `${endpoint}?${queryString}` : endpoint;
```
Comment (utsabc) [inline src/v0/destinations/tune/transform.js:55]: This is not needed as server adds the `params` as query params to url by default

File: src/v0/destinations/tune/transform.js [hunk 2]
```diff
@@ -8,6 +8,13 @@ const {
   isNotEmpty,
 } = require('../../util');
 
+const getTuneEndpoint = (subdomain) => `https://${subdomain}.go2cloud.org/aff_l`;
+
+const toQueryString = (params) =>
+  Object.entries(params)
+    .map(([key, value]) => `${encodeURIComponent(key)}=${encodeURIComponent(value)}`)
+    .join('&');
```
Comment (utsabc) [inline src/v0/destinations/tune/transform.js:16]: not needed

**Review by utsabc — CHANGES_REQUESTED** [review]

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3833/test-report.html

File: src/v0/destinations/tune/transform.js (hunk 1 above)
Comment (aanshi07) [reply src/v0/destinations/tune/transform.js:55]: @utsabc but then shouldn't it get reflected in the test case ?

Comment (utsabc) [reply src/v0/destinations/tune/transform.js:55]: we dont need that, server will handle from params you can test once in local

Comment (aanshi07) [reply src/v0/destinations/tune/transform.js]: done

**Review by krishna2020 — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Pull request merged**